resolver/resolver.cc

```diff
@@ -1388,18 +1391,96 @@ class ResolveSignaturesWalk {
                 Exception::raise("Wrong arg count");
             }
 
+            auto *memberType = core::cast_type<core::LambdaParam>(data->resultType.get());
+            ENFORCE(memberType != nullptr);
+
+            auto lowerBound = core::Types::bottom();
+            auto upperBound = core::Types::bottom();
+
+            // NOTE: the resultType is set back in the namer to be a LambdaParam
+            // with `T.untyped` for its bounds. We fix that here by setting the
+            // bounds to top and bottom.
+            memberType->lowerBound = core::Types::bottom();
+            memberType->upperBound = core::Types::top();
+
+            core::LambdaParam *parentType = nullptr;
+            auto parentMember = data->owner.data(ctx)->superClass().data(ctx)->findMember(ctx, data->name);
+            if (parentMember.exists()) {
+                parentType = core::cast_type<core::LambdaParam>(parentMember.data(ctx)->resultType.get());
+                ENFORCE(parentType != nullptr);
+            }
+
+            // Initialize the upper/lower bounds according to definition in the
+            // super class.
+            if (parentType != nullptr) {
+                memberType->lowerBound = parentType->lowerBound;
```

Comment:
This is magic, can we require users to spell it out?

Reply:
Yes, initializing to top and bottom should be sufficient here.

Reply:
Also, the bounds check WRT the super class should be done outside of the `if (hash)` block.

Reply:
I've simplified this in 8d5992c, could you take another look?